python/graphstorm/model/multitask_gnn.py

```diff
@@ -93,10 +101,102 @@ def __init__(self, alpha_l2norm):
         self._task_pool = {}
         self._decoder = nn.ModuleDict()
         self._loss_fn = nn.ModuleDict()
+        self._node_embed_norm_method = {}
         self._warn_printed = False
 
+    def normalize_task_node_embs(self, task_id, embs, inplace=False):
+        """ Normalize node embeddings when needed.
+
+            normalize_task_node_embs should be called when embs stores embeddings
+            of every node.
+
+        Parameters
+        ----------
+        task_id: str
+            Task ID.
+        embs: dict of Tensors
+            A dict of node embeddings.
+        inplace: bool
+            Whether to do inplace normalization.
+
+        Returns
+        -------
+        dict of Tensors:
+            Normalized node embeddings.
+        """
+        if self._node_embed_norm_method[task_id] is not None:
+            rank = get_rank()
+            world_size = get_world_size()
+            for key, emb in embs.items():
+                if isinstance(emb, (dgl.distributed.DistTensor, LazyDistTensor)):
+                    # If emb is a distributed tensor, multiple processes are doing
+                    # embdding normalization concurrently. We need to split
+                    # the task. (From full_graph_inference)
+                    start, end = get_data_range(rank, world_size, len(emb))
+                    new_embs = emb if inplace else \
```

Comment:
if inplace=False, may create a new dictionary, e.g., `new_embs = {}` and assign values like 
`new_emb = create_dist_tensor()`
`new_embs[key] = new_emb`

As the doc strings said, the `new_embs` should be a dictionary of tensors. But the line 149 shows it is a tensor.